File: backtestApp/metric_widget.py
import logging
from PyQt5.QtWidgets import QFrame, QVBoxLayout, QLabel, QSizePolicy
from PyQt5.QtGui import QFont

class MetricWidget(QFrame):
    """Widget pour afficher une métrique avec un titre, une valeur et une variation optionnelle"""
    def __init__(self, title, value="", delta="", delta_color="normal", parent=None):
        super().__init__(parent)
        self.setFrameShape(QFrame.StyledPanel)
        self.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Fixed)
        
        layout = QVBoxLayout(self)
        layout.setContentsMargins(10, 10, 10, 10)
        
        # Titre
        self.title_label = QLabel(title)
        title_font = QFont()
        title_font.setPointSize(10)
        self.title_label.setFont(title_font)
        
        # Valeur
        self.value_label = QLabel(value)
        value_font = QFont()
        value_font.setPointSize(12)
        value_font.setBold(True)
        self.value_label.setFont(value_font)
        
        # Delta
        self.delta_label = QLabel(delta)
        delta_font = QFont()
        delta_font.setPointSize(10)
        self.delta_label.setFont(delta_font)
        if delta_color == "normal":
            self.delta_label.setStyleSheet("color: green")
        elif delta_color == "inverse":
            self.delta_label.setStyleSheet("color: red")
        
        layout.addWidget(self.title_label)
        layout.addWidget(self.value_label)
        if delta:
            layout.addWidget(self.delta_label)
        
        self.setLayout(layout)
    
    def update_values(self, value, delta="", delta_color="normal"):
        self.value_label.setText(value)
        if delta:
            self.delta_label.setText(delta)
            if delta_color == "normal":
                self.delta_label.setStyleSheet("color: green")
            elif delta_color == "inverse":
                self.delta_label.setStyleSheet("color: red")

File: backtest_app_cpp/src/metric_widget.cpp
#include "metric_widget.h"
#include <QFont>
#include <QDebug>

MetricWidget::MetricWidget(const QString& title, const QString& value,
                           const QString& delta, const QString& delta_color,
                           QWidget* parent)
    : QFrame(parent)
{
    // Configuration du cadre
    setFrameShape(QFrame::StyledPanel);
    setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
    
    // Création du layout
    QVBoxLayout* layout = new QVBoxLayout(this);
    layout->setContentsMargins(10, 10, 10, 10);
    
    // Titre
    m_titleLabel = new QLabel(title, this);
    QFont titleFont;
    titleFont.setPointSize(10);
    m_titleLabel->setFont(titleFont);
    
    // Valeur
    m_valueLabel = new QLabel(value, this);
    QFont valueFont;
    valueFont.setPointSize(12);
    valueFont.setBold(true);
    m_valueLabel->setFont(valueFont);
    
    // Delta (variation)
    m_deltaLabel = new QLabel(delta, this);
    QFont deltaFont;
    deltaFont.setPointSize(10);
    m_deltaLabel->setFont(deltaFont);
    
    // Définir la couleur du delta
    if (delta_color == "normal") {
        m_deltaLabel->setStyleSheet("color: green");
    } else if (delta_color == "inverse") {
        m_deltaLabel->setStyleSheet("color: red");
    }
    
    // Ajout des widgets au layout
    layout->addWidget(m_titleLabel);
    layout->addWidget(m_valueLabel);
    if (!delta.isEmpty()) {
        layout->addWidget(m_deltaLabel);
    } else {
        m_deltaLabel->setVisible(false);
    }
}

void MetricWidget::updateValues(const QString& value, const QString& delta,
                                const QString& delta_color)
{
    // Mettre à jour la valeur
    m_valueLabel->setText(value);
    
    // Mettre à jour le delta s'il est fourni
    if (!delta.isEmpty()) {
        m_deltaLabel->setText(delta);
        m_deltaLabel->setVisible(true);
        
        // Mettre à jour la couleur du delta
        if (delta_color == "normal") {
            m_deltaLabel->setStyleSheet("color: green");
        } else if (delta_color == "inverse") {
            m_deltaLabel->setStyleSheet("color: red");
        }
    } else {
        m_deltaLabel->setVisible(false);
    }
}